### PR TITLE
tools/TestAllCalls.cpp: add lambda guards to test

### PR DESCRIPTION
**Pull request by ndellingwood**
guard test based on whether lambdas are enabled for the given execution space

### CODE REVIEW

**Comment by ndellingwood** [comment]
The if-guards for lambda support were added to address failing cuda + debug nightly builds on apollo; in cuda builds, lambda support is not enabled by default.
I also added an explicit `RangePolicy` to the kernel calls to prevent potential testing issues in cases like Cuda_Serial or Cuda_OpenMP during testing.

**Comment by ndellingwood** [comment]
For reference, link to the failing build:

https://jenkins-son.sandia.gov/job/Kokkos_apollo_debug/1189/console

Sample error message this addresses:
```bash
02:34:39 /home/jenkins/slave/workspace/Kokkos_apollo_debug/kokkos/core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp(447): error: The closure type for a lambda ("lambda [](int)->void", defined at /home/jenkins/slave/workspace/Kokkos_apollo_debug/kokkos/core/unit_test/tools/TestAllCalls.cpp:59) cannot be used in the template argument type of a __global__ function template instantiation, unless the lambda is defined within a __device__ or __global__ function, or the lambda is an 'extended lambda' and the flag --expt-extended-lambda is specified
02:34:39           detected during:
02:34:39             instantiation of "Kokkos::Impl::cuda_parallel_launch_local_memory" based on template argument <Kokkos::Impl::ParallelFor<lambda [](int)->void, Kokkos::RangePolicy<Kokkos::DefaultExecutionSpace>, Kokkos::CudaSpace::execution_space>> 
02:34:39 (447): here
02:34:39             instantiation of "cudaFuncAttributes Kokkos::Impl::CudaParallelLaunch<DriverType, Kokkos::LaunchBounds<0U, 0U>, Kokkos::Impl::Experimental::CudaLaunchMechanism::LocalMemory>::get_cuda_func_attributes() [with DriverType=Kokkos::Impl::ParallelFor<lambda [](int)->void, Kokkos::RangePolicy<Kokkos::DefaultExecutionSpace>, Kokkos::CudaSpace::execution_space>]" 
02:34:39 /home/jenkins/slave/workspace/Kokkos_apollo_debug/kokkos/core/src/Cuda/Kokkos_Cuda_Parallel.hpp(617): here
02:34:39             instantiation of "void Kokkos::Impl::ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::Cuda>::execute() const [with FunctorType=lambda [](int)->void, Traits=<Kokkos::DefaultExecutionSpace>]" 
02:34:39 /home/jenkins/slave/workspace/Kokkos_apollo_debug/kokkos/core/src/Kokkos_Parallel.hpp(215): here
02:34:39             instantiation of "void Kokkos::parallel_for(size_t, const FunctorType &, const std::__cxx11::string &) [with FunctorType=lambda [](int)->void]" 
02:34:39 /home/jenkins/slave/workspace/Kokkos_apollo_debug/kokkos/core/src/Kokkos_Parallel.hpp(232): here
02:34:39             instantiation of "void Kokkos::parallel_for(const std::__cxx11::string &, const ExecPolicy &, const FunctorType &) [with ExecPolicy=int, FunctorType=lambda [](int)->void]" 
02:34:39 /home/jenkins/slave/workspace/Kokkos_apollo_debug/kokkos/core/unit_test/tools/TestAllCalls.cpp(58): here
```

Reproducer instructions on apollo:
```bash
   #   Load modules:
         module load sems-env sems-cmake/3.12.2 kokkos-env kokkos-cuda/9.1 sems-gcc/5.3.0 kokkos-hwloc/1.10.1/base
 
   #   Use generate_makefile line below to call cmake which generates makefile for this build:
         $KOKKOS_PATH/generate_makefile.bash --with-devices=Cuda,OpenMP --arch=SNB,Volta70 --compiler=$KOKKOS_PATH/bin/nvcc_wrapper --cxxflags="-g -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="11" --ldflags="" --with-cuda --kokkos-path=$KOKKOS_PATH --no-examples --with-options=enable_large_mem_tests --debug
```

File: core/unit_test/tools/TestAllCalls.cpp [hunk 1]
```diff
@@ -50,19 +50,26 @@
...
-    Kokkos::View<int*> dst_view("destination", 10);
+    using execution_space = Kokkos::DefaultExecutionSpace;
+    using memory_space    = typename execution_space::memory_space;
+    Kokkos::View<int*, memory_space> src_view("source", 10);
+    Kokkos::View<int*, memory_space> dst_view("destination", 10);
     Kokkos::deep_copy(dst_view, src_view);
     Kokkos::parallel_for(
-        "parallel_for", 1, KOKKOS_LAMBDA(int i) { (void)i; });
+        "parallel_for",
+        Kokkos::RangePolicy<Kokkos::DefaultExecutionSpace>(0, 1),
```
Comment (masterleinad) [inline core/unit_test/tools/TestAllCalls.cpp:63]: You could replace `Kokkos::DefaultExecutionSpace` by `execution_space` here and below if we are already defining the alias.

Comment (ndellingwood) [reply core/unit_test/tools/TestAllCalls.cpp:63]: Good catch, I added this prior to the using statement and never went back to cleanup. Thanks!

Comment (ndellingwood) [reply core/unit_test/tools/TestAllCalls.cpp]: @masterleinad cleaned up like you suggested, reapplied clang-format and rebased for clean commit history. Should be good to go now.

Comment (masterleinad) [reply core/unit_test/tools/TestAllCalls.cpp]: Thanks!

**Comment by ndellingwood** [comment]
@crtrott @dalg24 ready for merge

**Pull request merged**